addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/service/NetworkUtils.java

```diff
@@ -56,13 +64,37 @@
                     }
                 }
             }
-        } catch (SocketException e) {
+        } catch (SocketException ignored) {
+            // If we are not allowed to enumerate, we return an empty result set.
         }
 
         return interfaceIPs;
     }
 
     /**
+     * Get a set of all interface names.
+     *
+     * @return Set of interface names
+     */
+    public static TreeSet<String> getInterfaceNames() {
```

Comment:
Can it return a `Set ` or maybe `SortedSet `or should it really be a `TreeSet`, see also next line